include/mata/nfa/nfa.hh

```diff
@@ -213,9 +213,11 @@ public:
     Nfa& concatenate(const Nfa& aut);
 
     /**
-     * @brief In-place union
+     * @brief In-place nondeterministic union with @p aut.
+     *
+     * Does not add epsilon transitions, just unites initial and final states.
      */
-    Nfa& uni(const Nfa &aut);
```

Comment:
I think there should still be some main simply-named union function (`uni`, `unionn`, `unite`, or whatever), because all other operations have such functions. It can even have parameters, like other functions (one now could be `preserve_determinism`), but maybe we do not want that.

Reply:
What would the function do, however? I think that the most basic "union" operation is the current `union_nondet()`. Therefore, the function would just call the `union_nondet()`, and if the parameter `preserve_determinism` was set, the function would just call `union_product()`. And the purpose of this whole PR was to remove the general `uni()` function. If we were to return it back, the naming problem would be here again. I personally am all for `unite()`, for example, but I am not sure what the purpose of such a function would be if the only thing the method does is calling other methods without any reasonable modifications.

Reply:
Ah, so the point of this PR was to remove the general function. Then yes, this solution is ok.

The purpose would be for the user to not have to think what function to use for union, if we want for him to always think and decide what to use, then this is ok. It would be very similar to how `is_lang_empty()` is implemented, by default it just calls one of the specialized emptiness test functions.

Reply:
Not necessarily to remove the general function, but to remove the naming problem for `union()` function which cannot be called union. We could maybe agree on adding a general function called `unite()`, or `lang_union()`, for example, which would call these two functions.

Reply:
@kilohsakul Any preferences?